ddl/index.go

```diff
@@ -49,6 +49,10 @@ func buildIndexColumns(columns []*model.ColumnInfo, idxColNames []*ast.IndexColN
 			return nil, errKeyColumnDoesNotExits.Gen("column does not exist: %s", ic.Column.Name)
 		}
 
+		if col.Flen == 0 {
```

Comment:
A statement as `create index idx on t (i)` will not pass table column type, it is not the same case of `create table` statement.